File: crates/sdkwork-rtc-core/src/provider_route.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct RtcProviderRoute {
    pub id: String,
    pub tenant_id: String,
    pub organization_id: String,
    pub provider_profile_id: String,
    pub route_type: String,
    pub region: Option<String>,
    pub priority: i32,
    pub status: RtcProviderRouteStatus,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum RtcProviderRouteStatus {
    Active,
    Disabled,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct RtcProviderRouteCommand {
    pub provider_profile_id: String,
    pub route_type: String,
    pub region: Option<String>,
    pub priority: i32,
    pub status: Option<RtcProviderRouteStatus>,
}

File: apps/sdkwork-rtc-flutter-mobile/packages/sdkwork_rtc_flutter_mobile_admin_core/lib/src/models/provider_route.dart
class ProviderRoute {
  final String id;
  final String tenantId;
  final String organizationId;
  final String providerProfileId;
  final String routeType;
  final String? region;
  final int priority;
  final String status;

  ProviderRoute({
    required this.id,
    required this.tenantId,
    required this.organizationId,
    required this.providerProfileId,
    required this.routeType,
    this.region,
    required this.priority,
    required this.status,
  });

  factory ProviderRoute.fromJson(Map<String, dynamic> json) {
    return ProviderRoute(
      id: json['id'] as String,
      tenantId: json['tenantId'] as String,
      organizationId: json['organizationId'] as String,
      providerProfileId: json['providerProfileId'] as String,
      routeType: json['routeType'] as String,
      region: json['region'] as String?,
      priority: json['priority'] as int,
      status: json['status'] as String,
    );
  }
}

class ProviderRouteCommand {
  final String providerProfileId;
  final String routeType;
  final String? region;
  final int priority;
  final String? status;

  ProviderRouteCommand({
    required this.providerProfileId,
    required this.routeType,
    this.region,
    required this.priority,
    this.status,
  });

  Map<String, dynamic> toJson() {
    return {
      'providerProfileId': providerProfileId,
      'routeType': routeType,
      if (region != null) 'region': region,
      'priority': priority,
      if (status != null) 'status': status,
    };
  }
}
